### PR TITLE
[stable/minio] Fixes Post-install-create-bucket-job fails on openshift

### LINKED ISSUE #20757: [stable/minio] Post-install-create-bucket-job fails on openshift
**Describe the bug**

Post-install-create-bucket-job fails because it uses `mc` to create the bucket, but `mc` needs write right to `$HOME`. On OpenShift deployments the container is run as a random user belonging to root group, therefore not having write right to `/`.
One solution would be the set `HOME=/tmp` for instance.

**Which chart**:
Minio

### PR DESCRIPTION
**Pull request by Jaydee94**
#### What this PR does / why we need it:
This PR adds a new value for optionally configuring the config-dir of the minIO-client.
The default config-dir (default: "/root/.mc") does not work on OpenShift.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #20757

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/minio/templates/_helper_create_bucket.txt [hunk 1]
```diff
@@ -1,6 +1,13 @@
 #!/bin/sh
 set -e ; # Have script exit in the event of a failed command.
 
+{{- if .Values.mcConfigDir }}
+MC_CONFIG_DIR="{{ .Values.mcConfigDir}}"
+MC_CALL="/usr/bin/mc ${MC_CONFIG_DIR}"
```
Comment (harshavardhana) [inline stable/minio/templates/_helper_create_bucket.txt]: you are missing `--config-dir`

File: stable/minio/templates/_helper_create_bucket.txt [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+{{- else }}
+MC_CALL="/usr/bin/mc"
```
Comment (harshavardhana) [inline stable/minio/templates/_helper_create_bucket.txt]: Just call this `MC` to be simple

```suggestion
MC="/usr/bin/mc"
```

File: stable/minio/README.md [hunk 3]
```diff
@@ -115,6 +115,7 @@ The following table lists the configurable parameters of the MinIO chart and the
 | `secretKey`                               | Default secret key (8 to 40 characters)                                                                                                 | `wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY` |
 | `certsPath`                               | Default certs path location                                                                                                             | `/etc/minio/certs`                         |
 | `configPathmc`                            | Default config file location for MinIO client - mc                                                                                      | `/etc/minio/mc`                            |
+| `mcConfigDir`                             | Customize the config-dir of the MinIO client - mc (default: "/root/.mc")                                                                | `""`                                       |
```
Comment (harshavardhana) [inline stable/minio/README.md]: This is not needed we already have `configPathmc` can that not be re-purposed?

File: stable/minio/templates/_helper_create_bucket.txt [hunk 4]
```diff
@@ -1,6 +1,13 @@
 #!/bin/sh
 set -e ; # Have script exit in the event of a failed command.
 
+{{- if .Values.mcConfigDir }}
```
Comment (harshavardhana) [inline stable/minio/templates/_helper_create_bucket.txt]: This should be `configPathmc` instead of `mcConfigDir`

File: stable/minio/values.yaml [hunk 5]
```diff
@@ -58,6 +58,7 @@ accessKey: "AKIAIOSFODNN7EXAMPLE"
 secretKey: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
 certsPath: "/etc/minio/certs/"
 configPathmc: "/etc/minio/mc/"
+mcConfigDir: ""
```
Comment (harshavardhana) [inline stable/minio/values.yaml]: As suggested `configPathmc` serves the same purpose unfortunately it was never used.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by Jaydee94** [comment]
@harshavardhana Thanks for youre review. I have added your suggestions and reused the variable configPathmc.

**Pull request merged**